lib/compat.php

```diff
@@ -25,22 +151,72 @@
 	 * @return WP_Block_Type|false The registered block type on success, or false on failure.
 	 */
 	function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
-		$file = ( substr( $file_or_folder, -10 ) !== 'block.json' ) ?
-			trailingslashit( $file_or_folder ) . 'block.json' :
+		$filename      = 'block.json';
+		$metadata_file = ( substr( $file_or_folder, -strlen( $filename ) ) !== $filename ) ?
+			trailingslashit( $file_or_folder ) . $filename :
 			$file_or_folder;
-		if ( ! file_exists( $file ) ) {
+		if ( ! file_exists( $metadata_file ) ) {
 			return false;
 		}
 
-		$metadata = json_decode( file_get_contents( $file ), true );
-		if ( ! is_array( $metadata ) ) {
+		$metadata = json_decode( file_get_contents( $metadata_file ), true );
+		if ( ! is_array( $metadata ) || empty( $metadata['name'] ) ) {
 			return false;
 		}
+		$metadata['file'] = $metadata_file;
+
+		$settings          = array();
+		$property_mappings = array(
+			'title'       => 'title',
+			'category'    => 'category',
```

Comment:
fix was just to add a property for `context` here. see https://github.com/WordPress/gutenberg/pull/23180

Reply:
We also need to add `providesContext` there.